lib/matplotlib/axes/_base.py

```diff
@@ -985,14 +1040,30 @@ def clear(self):
         """clear the axes"""
         self.cla()
 
+    def set_prop_cycle(self, prop_cycle):
+        """
+        Set the prop cycle for any future plot commands on this Axes.
+
+        *prop_cycle* is a :class:Cycler object.
+        Can also be `None` to reset to the cycle defined by the
+        current style.
+        """
+        self._get_lines.set_prop_cycle(prop_cycle)
+        self._get_patches_for_fill.set_prop_cycle(prop_cycle)
+
     def set_color_cycle(self, clist):
         """
         Set the color cycle for any future plot commands on this Axes.
 
         *clist* is a list of mpl color specifiers.
+
+        .. deprecated:: 1.5
```

Comment:
Why do we want this method to "blow away" property cyclers?  I think we should migrate fully over to property cyclers now.

1. In the ``rcsetup`` or somewhere around there, we should treat setting ``rcParam['axes.color_cycle']`` as a shortcut for setting ``rcParam['axes.prop_cycle']``.
2. This method, ``set_color_cycle``, should only set the **color** component of the property cycle, i.e.:
  a. If no property cycle exists, then set it as just a color cycle, otherwise...
  b. override the color part of the cycle, this might error for example if the ``color`` property had an inner-product with something else, and the new color-component of the cycle had a different length.  Also not sure about this part, but if the cycler didn't have a color property set, then we may want to throw an error saying that we weren't sure if we should set the color cycle using an inner product, or an outer product...